Feat/go open source

cloedu87 opened the pull request:
all things are going open source!

focus points:
- [x] license
- [x] repo config
- [x] environment config
- [x] clean up
- [x] github workflow
- [x] helm chart

cloedu87 committed:
feat(license): moved from GNU GPLv3 to GNU AGPLv3 to cover potential usage of forks as SAAS

cloedu87 committed:
chore: Merge remote-tracking branch 'origin/main' into feat/move-license-to-GNU-AGPLv3

cloedu87 committed:
Merge branch 'main' into feat/move-license-to-GNU-AGPLv3

cloedu87 committed:
feat: add zebbra release management to the repo

cloedu87 committed:
feat: license docs, contribution guides, basic ai context rules, and gh templates for open source readyness

cloedu87 committed:
feat: improve docs; make restrictions to run image builds and deployments only on owners (zebbra)
repository

cloedu87 committed:
chore: Merge remote-tracking branch 'origin/main' into feat/go-open-source

maetthu commented:
You can use security@zebbra.ch - CISO does not have a dedicated mail address :wink:

maetthu commented:
https://github.com/zebbra/data_aggregator/blob/main/helm/staging.yaml#L11 Needs to be addressed, but I guess we will talk about this in helm discussion. Other than that, LGTM

cloedu87 commented:
> https://github.com/zebbra/data_aggregator/blob/main/helm/staging.yaml#L11 Needs to be addressed, but I guess we will talk about this in helm discussion. Other than that, LGTM

ok thx! I have to clean up more things... like user seeds and stuff. then I'll continue.

cloedu87 committed:
feat: housekeeping on the way of making things opensource...

cloedu87 committed:
feat: because develop is now the default branch, it has to be respected
in ci workflow appropriatlely

cloedu87 committed:
fix: Add develop branch to CI/CD deploy pipeline

dependabot[bot] committed:
chore(deps-dev): bump git_hooks in the patch-updates group (#767)

Bumps the patch-updates group with 1 update: [git_hooks](https://github.com/qgadrian/elixir_git_hooks).


Updates `git_hooks` from 0.8.0 to 0.8.1
- [Release notes](https://github.com/qgadrian/elixir_git_hooks/releases)
- [Commits](https://github.com/qgadrian/elixir_git_hooks/compare/v0.8.0...v0.8.1)

---
updated-dependencies:
- dependency-name: git_hooks
  dependency-version: 0.8.1
  dependency-type: direct:development
  update-type: version-update:semver-patch
  dependency-group: patch-updates
...

Signed-off-by: dependabot[bot] <support@github.com>
Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

cloedu87 committed:
feat: some more docs

cloedu87 committed:
Merge branch 'main' into feat/go-open-source

dependabot[bot] committed:
chore(deps): bump the patch-updates group with 4 updates (#768)

Bumps the patch-updates group with 4 updates: [ash](https://github.com/ash-project/ash), [ash_authentication](https://github.com/team-alembic/ash_authentication), [ash_postgres](https://github.com/ash-project/ash_postgres) and [phoenix_live_view](https://github.com/phoenixframework/phoenix_live_view).


Updates `ash` from 3.5.10 to 3.5.11
- [Changelog](https://github.com/ash-project/ash/blob/main/CHANGELOG.md)
- [Commits](https://github.com/ash-project/ash/compare/v3.5.10...v3.5.11)

Updates `ash_authentication` from 4.8.6 to 4.8.7
- [Release notes](https://github.com/team-alembic/ash_authentication/releases)
- [Changelog](https://github.com/team-alembic/ash_authentication/blob/main/CHANGELOG.md)
- [Commits](https://github.com/team-alembic/ash_authentication/compare/v4.8.6...v4.8.7)

Updates `ash_postgres` from 2.5.19 to 2.5.20
- [Changelog](https://github.com/ash-project/ash_postgres/blob/main/CHANGELOG.md)
- [Commits](https://github.com/ash-project/ash_postgres/compare/v2.5.19...v2.5.20)

Updates `phoenix_live_view` from 1.0.11 to 1.0.12
- [Changelog](https://github.com/phoenixframework/phoenix_live_view/blob/v1.0.12/CHANGELOG.md)
- [Commits](https://github.com/phoenixframework/phoenix_live_view/compare/v1.0.11...v1.0.12)

---
updated-dependencies:
- dependency-name: ash
  dependency-version: 3.5.11
  dependency-type: direct:production
  update-type: version-update:semver-patch
  dependency-group: patch-updates
- dependency-name: ash_authentication
  dependency-version: 4.8.7
  dependency-type: direct:production
  update-type: version-update:semver-patch
  dependency-group: patch-updates
- dependency-name: ash_postgres
  dependency-version: 2.5.20
  dependency-type: direct:production
  update-type: version-update:semver-patch
  dependency-group: patch-updates
- dependency-name: phoenix_live_view
  dependency-version: 1.0.12
  dependency-type: direct:production
  update-type: version-update:semver-patch
  dependency-group: patch-updates
...

Signed-off-by: dependabot[bot] <support@github.com>
Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

mbaertschi committed:
fix(cicd): remove auto-merge

cloedu87 committed:
chore: minor docs

cloedu87 committed:
Merge branch 'main' into feat/go-open-source

cloedu87 committed:
feat: migrate helm chart to wrapper and dependency

cloedu87 committed:
feat: Remove deployment step from CI/CD workflow

cloedu87 committed:
feat: remove secret-env reference, 'cause this will be handled in the deployment repository

cloedu87 committed:
chore: merge in origin/develop

cloedu87 committed:
chore: merge in origin/develop

cloedu87 committed:
chore: Merge remote-tracking branch 'origin/develop' into feat/go-open-source

cloedu87 committed:
fix: make references automappable #820; dependency updates

cloedu87 committed:
fix: update dependencies and fix issues w/ login loop; write little test
for this; dark / light login logo config overwrite

cloedu87 committed:
chore: dependency updates and clean ups; fix linting;

cloedu87 committed:
chore: Merge remote-tracking branch 'origin/develop' into feat/go-open-source

cloedu87 committed:
feat: deployment and values config for argo cd deployment

cloedu87 committed:
feat: add missed keys and checked rendering

cloedu87 committed:
fix: run helm chart deployment on all breanches

cloedu87 committed:
fix: add correct app and chart version

cloedu87 committed:
fix: typo

cloedu87 committed:
fix: use .Values.env in deployment

cloedu87 committed:
feat: make annotations for migration/jobs overridable

cloedu87 committed:
feat: updated helm default values

cloedu87 committed:
feat: deployment isn't managed by helm anymore

cloedu87 committed:
chore: chart renaming data_aggregator to aggregator

cloedu87 committed:
fix: separate job annotations from migration specific annotations for
override reasons in deployment repo

cloedu87 committed:
fix: deployment and migrate env handling

cloedu87 committed:
fix: fu typo

cloedu87 committed:
fix: add env var reference to catalog init container

cloedu87 committed:
fix: template rendering with empty secret vars

hwuethrich commented on .github/workflows/cicd.yml:
```diff
@@ -102,6 +105,7 @@ jobs:
     name: Build Docker image
     needs: test
     runs-on: hetzner
+    if: github.repository == 'zebbra/data_aggregator'
```
Is this required? It will not run anyway because of "hetzner"

cloedu87 replied:
actually not necessary, but I'll keep it to have it more explicitly configured. if that's fine for you.

cloedu87 committed:
Feat/publicaton adjustments and text blocks (#830)

* feat: adjust publication modal, textblocks and handling

* fix: iucn filter, fix filtering according to encoded_record data, not
record data

* feat: remove unwanted sorting headers

* feat: solves #844 presentation and fallback of elevation attributes in
list and sidepanel

* feat: make elevation beautiful

* fix: on cancel im port refresh collection records count, solves #845

* feat: remove geology from collection types, solves #846

* feat: info text while images are being extracted, solves #847

* fix: fixes from pr review

cloedu87 committed:
feat: deployment repo dispatch github action

cloedu87 committed:
fix: trigger helm chart publish

cloedu87 committed:
feat: make app version available from git tag and bake it into image;
simplify push and buit to one stepp; update helm chart version and
appVersion to the git tag value while building and push it back to the
repo

cloedu87 committed:
fix: run helm only if ci cd workflow passed alredy successful

cloedu87 committed:
fix: missing version number while compileing

cloedu87 committed:
fix: version var usable in docker build in compile task

cloedu87 committed:
feat: only modify helm chart, if it is a version, which is usable to
have injected to the elixir compile and docker build process

cloedu87 committed:
fix: text

maetthu committed:
fix: add missing hpa resource

maetthu committed:
fix: add default value for port
